andela/ah-shakas
# #160577487 List all users

#### What does this PR do?
- Have an authorised user view profiles of other users.
- creates a test folder for profiles' app and put all test files for the application

#### Description of Task to be completed?
- Have the following endpoints working
```
GET /api/profiles/
```
#### How should this be manually tested?
- After cloning [this repo](https://github.com/andela/ah-shakas), cd into the directory, checkout to branch ft-list-users-160577487 and run the application,
```git checkout ft-list-users-160577487```
```pipenv install```
```pipenv shell```
```pipenv install```
```
python manage.py migrate
```
```
python manage.py runserver
```
- using postman or any API client, test the endpoint with header:
_key_: Authorization, _value_:Token from  ```POST: /api/users/login/``` with payload in the format 
```
{
	"user": {
		"email": "testinguser@test.com",
		"username": "testinguser",
		"password": "testinguser123"
	}
}
```
- Note that you have to be registered with your account activated. The endpoint for registeration is 
```POST: /api/users/``` with payload in the format 
```
{
	"user": {
		"email": "testinguser@test.com",
		"username": "testinguser",
		"password": "testinguser123"
	}
}
```
#### Any background context you want to provide?
- Tests for this feature is found in authors/apps/profiles/test/test_view_all_profiles.py
- The implementation of this feature was largely dependant on 
https://www.pivotaltracker.com/story/show/160577479
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/160577487
#### Screenshots (if appropriate)
<img width="1122" alt="screen shot 2018-10-12 at 16 03 00" src="https://user-images.githubusercontent.com/39160236/46870809-5a1adb00-ce38-11e8-88f4-da287374c25a.png">
